compression/lz77.py

```diff
@@ -43,21 +57,14 @@ def __init__(self, window_size: int = 13, lookahead_buffer_size: int = 6) -> Non
         self.lookahead_buffer_size = lookahead_buffer_size
         self.search_buffer_size = self.window_size - self.lookahead_buffer_size
 
-    def compress(self, text: str) -> list:
+    def compress(self, text: str) -> List[Token]:
         """This method compresses given string text using LZ77 compression algorithm.
 
         Args:
             text (str): string that's going to be compressed
 
         Returns:
-            output (list): the compressed text
-
-        Tests:
-            >>> lz77_compressor = LZ77Compressor(13, 6)
-            >>> lz77_compressor.compress("ababcbababaa")
-            [(0, 0, 'a'), (0, 0, 'b'), (2, 2, 'c'), (4, 3, 'a'), (2, 2, 'a')]
-            >>> lz77_compressor.compress("aacaacabcabaaac")
-            [(0, 0, 'a'), (1, 1, 'c'), (3, 4, 'b'), (3, 3, 'a'), (1, 2, 'c')]
+            output (List[Token]): the compressed text
```

Comment:
Let's keep the tests!

Dataclass will create a `__str__()` method for you with output like `Token(offset=0, length=0, char='a')` so...
```suggestion
            output (List[Token]): the compressed text

        >>> lz77_compressor = LZ77Compressor(13, 6)
        >>> lz77_compressor.compress("ababcbababaa")  # doctest: +NORMALIZE_WHITESPACE
        [Token(offset=0, length=0, char='a'), Token(offset=0, length=0, char='b'),
         Token(offset=2, length=2, char='c'), Token(offset=4, length=3, char='a')
         Token(offset=2, length=2, char='a')]
```
Or you can define your own `__str__()` method to return `(0, 0, 'a')`.

Reply:
great idea, done